numpy/ma/core.pyi

```diff
@@ -1205,10 +1205,25 @@ identity: _convert2ma
 indices: _convert2ma
 ones: _convert2ma
 ones_like: _convert2ma
-squeeze: _convert2ma
+
+@overload
+def squeeze(
+    a: _ScalarT,
+    axis: _ShapeLike | None = ...,
+) -> _ScalarT: ...
+@overload
+def squeeze(
+    a: _ArrayLike[_ScalarT],
+    axis: _ShapeLike | None = ...,
+) -> _MaskedArray[_ScalarT]: ...
+@overload
+def squeeze(
+    a: ArrayLike,
+    axis: _ShapeLike | None = ...,
+) -> _MaskedArray[Any]: ...
```

Comment:
So it turns out that stubtest doesn't accept these being actual functions, and instead requires them to be instances of `numpy.ma.core._convert2ma`. You can see this in https://github.com/numpy/numtype/pull/468. It's currently a WIP, but there I (attempted to) explain how the `_convert2ma` can be made generic on it's `__call__` signature, so that a  callable `Protocol` can be used to specify the signature. It's a bit messy, but I'm afraid that it's the only way that stubtest well be able to swallow it. 

Are you willing to attempt doing this here as well, or would you prefer that we address this at a later point?